app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt

```diff
@@ -589,6 +591,41 @@ class MainActivity : SimpleActivity(), RefreshRecyclerViewListener {
         }
     }
 
+    private fun addBirthdaysAnniversariesAtStart() {
+        if (!hasPermission(PERMISSION_READ_CONTACTS)) {
```

Comment:
you should check config.addAnniversariesAutomatically and config.addBirthdaysAutomatically first, those are a lot quicker and simpler checks than some permissions